sky/serve/load_balancer.py

```diff
@@ -70,30 +74,110 @@ def _sync_with_controller(self):
+            # entering this function. In that case we will return an error here
+            # and retry to find next ready replica.
+            client = self._client_pool.get(url, None)
+            if client is None:
+                return RuntimeError(f'Client for {url} not found.')
+            worker_url = httpx.URL(path=request.url.path,
+                                   query=request.url.query.encode('utf-8'))
+            proxy_request = client.build_request(
+                request.method,
+                worker_url,
+                headers=request.headers.raw,
+                content=await request.body(),
+                timeout=constants.LB_STREAM_TIMEOUT)
+            proxy_response = await client.send(proxy_request, stream=True)
+            return fastapi.responses.StreamingResponse(
+                content=proxy_response.aiter_raw(),
+                status_code=proxy_response.status_code,
+                headers=proxy_response.headers,
+                background=background.BackgroundTask(proxy_response.aclose))
+        except (httpx.RequestError, httpx.HTTPStatusError) as e:
+            logger.error(f'Error when proxy request to {url}: '
+                         f'{common_utils.format_exception(e)}')
+            return e
+
+    async def _proxy_with_retries(
+            self, request: fastapi.Request) -> fastapi.responses.Response:
+        """Try to proxy the request to the endpoint replica with retries."""
+        self._request_aggregator.add(request)
+        # TODO(tian): Finetune backoff parameters.
+        backoff = common_utils.Backoff(initial_backoff=1)
+        # SkyServe supports serving on Spot Instances. To avoid preemptions
+        # during request handling, we add a retry here.
+        retry_cnt = 0
+        while True:
+            retry_cnt += 1
+            with self._lock:
+                ready_replica_url = self._load_balancing_policy.select_replica(
+                    request)
+            if ready_replica_url is None:
+                raise fastapi.HTTPException(
+                    # 503 means that the server is currently
+                    # unable to handle the incoming requests.
+                    status_code=503,
+                    detail='No ready replicas. '
+                    'Use "sky serve status [SERVICE_NAME]" '
+                    'to check the replica status.')
+            response_or_exception = await self._proxy_request_to(
+                ready_replica_url, request)
+            if not isinstance(response_or_exception, Exception):
+                return response_or_exception
+            # TODO(tian): Fail fast for errors like 404 not found.
+            if retry_cnt == constants.LB_MAX_RETRY:
+                exception = common_utils.format_exception(
+                    response_or_exception,
+                    use_bracket=True,
+                    brighten_error_class=False)
```

Comment:
nit: Instead of adding a new argument for this function, we can also use `common_utils.remove_color` to remove those color chars.

Reply:
Good point! Updated. Thanks!